==2903== HEAP SUMMARY:
==2903==     in use at exit: 46 bytes in 3 blocks
==2903==   total heap usage: 16 allocs, 13 frees, 1,346 bytes allocated
==2903== 
==2903== Searching for pointers to 3 not-freed blocks
==2903== Checked 584,440 bytes
==2903== 
==2903== 1 bytes in 1 blocks are definitely lost in loss record 1 of 3
==2903==    at 0x483B7F3: malloc (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_memcheck-amd64-linux.so)
==2903==    by 0x40185D: ft_calloc (get_next_line_utils.c:49)
==2903==    by 0x401536: read_file (get_next_line.c:77)
==2903==    by 0x40166A: get_next_line (get_next_line.c:110)
==2903==    by 0x4011C0: main (main.c:14)
==2903== 
==2903== 2 bytes in 1 blocks are definitely lost in loss record 2 of 3
==2903==    at 0x483B7F3: malloc (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_memcheck-amd64-linux.so)
==2903==    by 0x40185D: ft_calloc (get_next_line_utils.c:49)
==2903==    by 0x4019AA: ft_strdup (get_next_line_utils.c:86)
==2903==    by 0x401709: get_next_line (get_next_line.c:116)
==2903==    by 0x4011C0: main (main.c:14)
==2903== 
==2903== 43 bytes in 1 blocks are definitely lost in loss record 3 of 3
==2903==    at 0x483B7F3: malloc (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_memcheck-amd64-linux.so)
==2903==    by 0x40185D: ft_calloc (get_next_line_utils.c:49)
==2903==    by 0x4012C1: ft_strjoin_gnl (get_next_line.c:23)
==2903==    by 0x4015DB: read_file (get_next_line.c:93)
==2903==    by 0x40166A: get_next_line (get_next_line.c:110)
==2903==    by 0x4011C0: main (main.c:14)
==2903== 
==2903== LEAK SUMMARY:
==2903==    definitely lost: 46 bytes in 3 blocks
==2903==    indirectly lost: 0 bytes in 0 blocks
==2903==      possibly lost: 0 bytes in 0 blocks
==2903==    still reachable: 0 bytes in 0 blocks
==2903==         suppressed: 0 bytes in 0 blocks
==2903== 
==2903== ERROR SUMMARY: 23 errors from 13 contexts (suppressed: 0 from 0)
==2903== 
==2903== 1 errors in context 1 of 13:
==2903== Invalid read of size 1
==2903==    at 0x401316: ft_strjoin_gnl (get_next_line.c:28)
==2903==    by 0x4015DB: read_file (get_next_line.c:93)
==2903==    by 0x40166A: get_next_line (get_next_line.c:110)
==2903==    by 0x40123E: main (main.c:23)
==2903==  Address 0x4a4d860 is 0 bytes inside a block of size 2 free'd
==2903==    at 0x483CA3F: free (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_memcheck-amd64-linux.so)
==2903==    by 0x40175F: ft_free (get_next_line_utils.c:19)
==2903==    by 0x401723: get_next_line (get_next_line.c:118)
==2903==    by 0x401214: main (main.c:20)
==2903==  Block was alloc'd at
==2903==    at 0x483B7F3: malloc (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_memcheck-amd64-linux.so)
==2903==    by 0x40185D: ft_calloc (get_next_line_utils.c:49)
==2903==    by 0x4012C1: ft_strjoin_gnl (get_next_line.c:23)
==2903==    by 0x4015DB: read_file (get_next_line.c:93)
==2903==    by 0x40166A: get_next_line (get_next_line.c:110)
==2903==    by 0x401214: main (main.c:20)
==2903== 
==2903== 
==2903== 1 errors in context 2 of 13:
==2903== Invalid read of size 1
==2903==    at 0x401316: ft_strjoin_gnl (get_next_line.c:28)
==2903==    by 0x4015DB: read_file (get_next_line.c:93)
==2903==    by 0x40166A: get_next_line (get_next_line.c:110)
==2903==    by 0x401214: main (main.c:20)
==2903==  Address 0x4a4d750 is 0 bytes inside a block of size 2 free'd
==2903==    at 0x483CA3F: free (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_memcheck-amd64-linux.so)
==2903==    by 0x40175F: ft_free (get_next_line_utils.c:19)
==2903==    by 0x401723: get_next_line (get_next_line.c:118)
==2903==    by 0x4011EA: main (main.c:17)
==2903==  Block was alloc'd at
==2903==    at 0x483B7F3: malloc (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_memcheck-amd64-linux.so)
==2903==    by 0x40185D: ft_calloc (get_next_line_utils.c:49)
==2903==    by 0x4012C1: ft_strjoin_gnl (get_next_line.c:23)
==2903==    by 0x4015DB: read_file (get_next_line.c:93)
==2903==    by 0x40166A: get_next_line (get_next_line.c:110)
==2903==    by 0x4011EA: main (main.c:17)
==2903== 
==2903== 
==2903== 2 errors in context 3 of 13:
==2903== Invalid read of size 1
==2903==    at 0x48436E0: mempcpy (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_memcheck-amd64-linux.so)
==2903==    by 0x48EA7B1: _IO_new_file_xsputn (fileops.c:1236)
==2903==    by 0x48EA7B1: _IO_file_xsputn@@GLIBC_2.2.5 (fileops.c:1197)
==2903==    by 0x48D227B: __vfprintf_internal (vfprintf-internal.c:1688)
==2903==    by 0x48BCEBE: printf (printf.c:33)
==2903==    by 0x40138F: ft_strjoin_gnl (get_next_line.c:38)
==2903==    by 0x4015DB: read_file (get_next_line.c:93)
==2903==    by 0x40166A: get_next_line (get_next_line.c:110)
==2903==    by 0x401214: main (main.c:20)
==2903==  Address 0x4a4d750 is 0 bytes inside a block of size 2 free'd
==2903==    at 0x483CA3F: free (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_memcheck-amd64-linux.so)
==2903==    by 0x40175F: ft_free (get_next_line_utils.c:19)
==2903==    by 0x401723: get_next_line (get_next_line.c:118)
==2903==    by 0x4011EA: main (main.c:17)
==2903==  Block was alloc'd at
==2903==    at 0x483B7F3: malloc (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_memcheck-amd64-linux.so)
==2903==    by 0x40185D: ft_calloc (get_next_line_utils.c:49)
==2903==    by 0x4012C1: ft_strjoin_gnl (get_next_line.c:23)
==2903==    by 0x4015DB: read_file (get_next_line.c:93)
==2903==    by 0x40166A: get_next_line (get_next_line.c:110)
==2903==    by 0x4011EA: main (main.c:17)
==2903== 
==2903== 
==2903== 2 errors in context 4 of 13:
==2903== Invalid read of size 1
==2903==    at 0x48EA88C: _IO_new_file_xsputn (fileops.c:1219)
==2903==    by 0x48EA88C: _IO_file_xsputn@@GLIBC_2.2.5 (fileops.c:1197)
==2903==    by 0x48D227B: __vfprintf_internal (vfprintf-internal.c:1688)
==2903==    by 0x48BCEBE: printf (printf.c:33)
==2903==    by 0x40138F: ft_strjoin_gnl (get_next_line.c:38)
==2903==    by 0x4015DB: read_file (get_next_line.c:93)
==2903==    by 0x40166A: get_next_line (get_next_line.c:110)
==2903==    by 0x401214: main (main.c:20)
==2903==  Address 0x4a4d750 is 0 bytes inside a block of size 2 free'd
==2903==    at 0x483CA3F: free (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_memcheck-amd64-linux.so)
==2903==    by 0x40175F: ft_free (get_next_line_utils.c:19)
==2903==    by 0x401723: get_next_line (get_next_line.c:118)
==2903==    by 0x4011EA: main (main.c:17)
==2903==  Block was alloc'd at
==2903==    at 0x483B7F3: malloc (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_memcheck-amd64-linux.so)
==2903==    by 0x40185D: ft_calloc (get_next_line_utils.c:49)
==2903==    by 0x4012C1: ft_strjoin_gnl (get_next_line.c:23)
==2903==    by 0x4015DB: read_file (get_next_line.c:93)
==2903==    by 0x40166A: get_next_line (get_next_line.c:110)
==2903==    by 0x4011EA: main (main.c:17)
==2903== 
==2903== 
==2903== 2 errors in context 5 of 13:
==2903== Invalid read of size 1
==2903==    at 0x483EF54: strlen (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_memcheck-amd64-linux.so)
==2903==    by 0x48D3E94: __vfprintf_internal (vfprintf-internal.c:1688)
==2903==    by 0x48BCEBE: printf (printf.c:33)
==2903==    by 0x40138F: ft_strjoin_gnl (get_next_line.c:38)
==2903==    by 0x4015DB: read_file (get_next_line.c:93)
==2903==    by 0x40166A: get_next_line (get_next_line.c:110)
==2903==    by 0x401214: main (main.c:20)
==2903==  Address 0x4a4d751 is 1 bytes inside a block of size 2 free'd
==2903==    at 0x483CA3F: free (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_memcheck-amd64-linux.so)
==2903==    by 0x40175F: ft_free (get_next_line_utils.c:19)
==2903==    by 0x401723: get_next_line (get_next_line.c:118)
==2903==    by 0x4011EA: main (main.c:17)
==2903==  Block was alloc'd at
==2903==    at 0x483B7F3: malloc (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_memcheck-amd64-linux.so)
==2903==    by 0x40185D: ft_calloc (get_next_line_utils.c:49)
==2903==    by 0x4012C1: ft_strjoin_gnl (get_next_line.c:23)
==2903==    by 0x4015DB: read_file (get_next_line.c:93)
==2903==    by 0x40166A: get_next_line (get_next_line.c:110)
==2903==    by 0x4011EA: main (main.c:17)
==2903== 
==2903== 
==2903== 2 errors in context 6 of 13:
==2903== Invalid read of size 1
==2903==    at 0x483EF46: strlen (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_memcheck-amd64-linux.so)
==2903==    by 0x48D3E94: __vfprintf_internal (vfprintf-internal.c:1688)
==2903==    by 0x48BCEBE: printf (printf.c:33)
==2903==    by 0x40138F: ft_strjoin_gnl (get_next_line.c:38)
==2903==    by 0x4015DB: read_file (get_next_line.c:93)
==2903==    by 0x40166A: get_next_line (get_next_line.c:110)
==2903==    by 0x401214: main (main.c:20)
==2903==  Address 0x4a4d750 is 0 bytes inside a block of size 2 free'd
==2903==    at 0x483CA3F: free (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_memcheck-amd64-linux.so)
==2903==    by 0x40175F: ft_free (get_next_line_utils.c:19)
==2903==    by 0x401723: get_next_line (get_next_line.c:118)
==2903==    by 0x4011EA: main (main.c:17)
==2903==  Block was alloc'd at
==2903==    at 0x483B7F3: malloc (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_memcheck-amd64-linux.so)
==2903==    by 0x40185D: ft_calloc (get_next_line_utils.c:49)
==2903==    by 0x4012C1: ft_strjoin_gnl (get_next_line.c:23)
==2903==    by 0x4015DB: read_file (get_next_line.c:93)
==2903==    by 0x40166A: get_next_line (get_next_line.c:110)
==2903==    by 0x4011EA: main (main.c:17)
==2903== 
==2903== 
==2903== 2 errors in context 7 of 13:
==2903== Invalid read of size 1
==2903==    at 0x401301: ft_strjoin_gnl (get_next_line.c:26)
==2903==    by 0x4015DB: read_file (get_next_line.c:93)
==2903==    by 0x40166A: get_next_line (get_next_line.c:110)
==2903==    by 0x40123E: main (main.c:23)
==2903==  Address 0x4a4d860 is 0 bytes inside a block of size 2 free'd
==2903==    at 0x483CA3F: free (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_memcheck-amd64-linux.so)
==2903==    by 0x40175F: ft_free (get_next_line_utils.c:19)
==2903==    by 0x401723: get_next_line (get_next_line.c:118)
==2903==    by 0x401214: main (main.c:20)
==2903==  Block was alloc'd at
==2903==    at 0x483B7F3: malloc (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_memcheck-amd64-linux.so)
==2903==    by 0x40185D: ft_calloc (get_next_line_utils.c:49)
==2903==    by 0x4012C1: ft_strjoin_gnl (get_next_line.c:23)
==2903==    by 0x4015DB: read_file (get_next_line.c:93)
==2903==    by 0x40166A: get_next_line (get_next_line.c:110)
==2903==    by 0x401214: main (main.c:20)
==2903== 
==2903== 
==2903== 2 errors in context 8 of 13:
==2903== Invalid read of size 1
==2903==    at 0x401796: ft_strlen (get_next_line_utils.c:31)
==2903==    by 0x40129F: ft_strjoin_gnl (get_next_line.c:23)
==2903==    by 0x4015DB: read_file (get_next_line.c:93)
==2903==    by 0x40166A: get_next_line (get_next_line.c:110)
==2903==    by 0x401214: main (main.c:20)
==2903==  Address 0x4a4d750 is 0 bytes inside a block of size 2 free'd
==2903==    at 0x483CA3F: free (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_memcheck-amd64-linux.so)
==2903==    by 0x40175F: ft_free (get_next_line_utils.c:19)
==2903==    by 0x401723: get_next_line (get_next_line.c:118)
==2903==    by 0x4011EA: main (main.c:17)
==2903==  Block was alloc'd at
==2903==    at 0x483B7F3: malloc (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_memcheck-amd64-linux.so)
==2903==    by 0x40185D: ft_calloc (get_next_line_utils.c:49)
==2903==    by 0x4012C1: ft_strjoin_gnl (get_next_line.c:23)
==2903==    by 0x4015DB: read_file (get_next_line.c:93)
==2903==    by 0x40166A: get_next_line (get_next_line.c:110)
==2903==    by 0x4011EA: main (main.c:17)
==2903== 
==2903== 
==2903== 2 errors in context 9 of 13:
==2903== Invalid read of size 1
==2903==    at 0x401301: ft_strjoin_gnl (get_next_line.c:26)
==2903==    by 0x4015DB: read_file (get_next_line.c:93)
==2903==    by 0x40166A: get_next_line (get_next_line.c:110)
==2903==    by 0x401214: main (main.c:20)
==2903==  Address 0x4a4d750 is 0 bytes inside a block of size 2 free'd
==2903==    at 0x483CA3F: free (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_memcheck-amd64-linux.so)
==2903==    by 0x40175F: ft_free (get_next_line_utils.c:19)
==2903==    by 0x401723: get_next_line (get_next_line.c:118)
==2903==    by 0x4011EA: main (main.c:17)
==2903==  Block was alloc'd at
==2903==    at 0x483B7F3: malloc (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_memcheck-amd64-linux.so)
==2903==    by 0x40185D: ft_calloc (get_next_line_utils.c:49)
==2903==    by 0x4012C1: ft_strjoin_gnl (get_next_line.c:23)
==2903==    by 0x4015DB: read_file (get_next_line.c:93)
==2903==    by 0x40166A: get_next_line (get_next_line.c:110)
==2903==    by 0x4011EA: main (main.c:17)
==2903== 
==2903== 
==2903== 4 errors in context 10 of 13:
==2903== Invalid read of size 1
==2903==    at 0x4017B9: ft_strlen (get_next_line_utils.c:33)
==2903==    by 0x40129F: ft_strjoin_gnl (get_next_line.c:23)
==2903==    by 0x4015DB: read_file (get_next_line.c:93)
==2903==    by 0x40166A: get_next_line (get_next_line.c:110)
==2903==    by 0x401214: main (main.c:20)
==2903==  Address 0x4a4d750 is 0 bytes inside a block of size 2 free'd
==2903==    at 0x483CA3F: free (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_memcheck-amd64-linux.so)
==2903==    by 0x40175F: ft_free (get_next_line_utils.c:19)
==2903==    by 0x401723: get_next_line (get_next_line.c:118)
==2903==    by 0x4011EA: main (main.c:17)
==2903==  Block was alloc'd at
==2903==    at 0x483B7F3: malloc (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_memcheck-amd64-linux.so)
==2903==    by 0x40185D: ft_calloc (get_next_line_utils.c:49)
==2903==    by 0x4012C1: ft_strjoin_gnl (get_next_line.c:23)
==2903==    by 0x4015DB: read_file (get_next_line.c:93)
==2903==    by 0x40166A: get_next_line (get_next_line.c:110)
==2903==    by 0x4011EA: main (main.c:17)
==2903== 
==2903== ERROR SUMMARY: 23 errors from 13 contexts (suppressed: 0 from 0)
